### PR TITLE
[GPU] Optimize DepthToSpace

### PR DESCRIPTION
**Pull request by tuxedcat**
### Details:
 - Optimize ref kernel of DepthToSpace

### Tickets:
 - 78851

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/depth_to_space/depth_to_space_kernel_ref.cpp [hunk 1]
```diff
@@ -40,8 +40,15 @@ CommonDispatchData DepthToSpaceKernelRef::SetDefault(const depth_to_space_params
...
 
-    // this kernel only supports bfyx and b_fs_yx_fsv16 layout.
+    //The reason why reverse input/output of GetOptimalLocalWorkGroupSizes():
+    //Large X*Y*Z lws size is better than large batch lws, but current GetOptimalLocalWorkGroupSizes not work like that.
+    reverse(dims_by_gws.begin(), dims_by_gws.end());
+    reverse(dispatchData.gws.begin(), dispatchData.gws.end());
     dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo, in_layout, out_layout, dims_by_gws);
+    reverse(dispatchData.lws.begin(), dispatchData.lws.end());
+    reverse(dispatchData.gws.begin(), dispatchData.gws.end());
+    reverse(dims_by_gws.begin(), dims_by_gws.end());
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/depth_to_space/depth_to_space_kernel_ref.cpp:50]: dims_by_gws is local variable. No need to reverse it back.

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/depth_to_space/depth_to_space_kernel_ref.cpp [hunk 2]
```diff
@@ -40,8 +40,15 @@ CommonDispatchData DepthToSpaceKernelRef::SetDefault(const depth_to_space_params
                          params.outputs[0].Feature().v,
                          params.outputs[0].Z().v * params.outputs[0].Y().v * params.outputs[0].X().v };
 
-    // this kernel only supports bfyx and b_fs_yx_fsv16 layout.
+    //The reason why reverse input/output of GetOptimalLocalWorkGroupSizes():
+    //Large X*Y*Z lws size is better than large batch lws, but current GetOptimalLocalWorkGroupSizes not work like that.
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/depth_to_space/depth_to_space_kernel_ref.cpp:44]: nit: whitespace between `//` and actual comment.

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/depth_to_space/depth_to_space_kernel_ref.cpp (hunk 1 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/depth_to_space/depth_to_space_kernel_ref.cpp]: Done.

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/depth_to_space/depth_to_space_kernel_ref.cpp (hunk 2 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/depth_to_space/depth_to_space_kernel_ref.cpp]: Done.

**Review by isanghao — APPROVED** [review]

**Pull request merged**